File: pset6/readability/readability.py
from cs50 import get_string
s = get_string("Text: ")
letters = 0
words = 1
sentences = 0
for c in s:
    if c.islower():
        letters += 1
    if c.isupper():
        letters += 1
    if c.isspace():
        words += 1
    if c == '.' or c == '!' or c == '?':
        sentences += 1

L = float(letters * 100) / float(words)
S = float(sentences * 100) / float(words)
index = 0.000
index = (0.0588 * float(L)) - (0.296 * float(S)) - 15.8
if index >= 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {round(index)}")

File: pset2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string s = get_string("Text: ");
    int letters = 0;
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if isupper((s[i]))
        {
            letters += 1;
        }
        if islower((s[i]))
        {
            letters += 1;
        }
    }
    int words = 1;
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if isspace((s[i]))
        {
            words += 1;
        }
    }
    int sentences = 0;
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if (s[i] == '.' ||  s[i] == '!' || s[i] == '?')
        {
            sentences += 1;
        }
    }
    float L = (float) letters * 100 / (float) words;
    float S = (float) sentences * 100 / (float) words;
    float index;
    index = (0.0588 * (float) L) - (0.296 * (float) S) - 15.8;
    if (index >= 16)
    {
        printf("Grade 16+");
    }
    else if (index < 1)
    {
        printf("Before Grade 1");
    }
    else
    {
        printf("Grade %.f", round(index));
    }
    printf("\n");
}